test/unit/tt_sigaction.c

```diff
@@ -0,0 +1,79 @@
+#include "unit.h"
+
+#include <pthread.h>
+#include <stdbool.h>
+#include <string.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include "tt_sigaction.h"
+#include "clock.h"
+
+/** Test duration in seconds. */
+#define TEST_LEN 1.5
+/** Signal sending period in usec. Must be lower than second. */
+#define SIGNAL_PERIOD 1e3
+
+pthread_t main_thread;
+
+void *
+thread_f(void *arg)
+{
+	double start = clock_monotonic();
+	while (clock_monotonic() - start < TEST_LEN)
+		sched_yield();
```

Comment:
We cannot sleep because sleep implementation can use SIGALRM, which is used by test. Changed to busy cycle.

Reply:
What wrong can happen in this case? We use `sleep` here and there so avoiding it in the test looks suspicious. You could use `thread_sleep` - it calls `nanosleep` under the hood. There's nothing said about `alarm` in `nanosleep` man page.

Reply:
Busy loop is no better than `sched_yield`.

Reply:
Changed to `usleep`.

Reply:
Still, why not use `thread_sleep` instead of

```C
	double start = clock_monotonic();
	while (clock_monotonic() - start < TEST_LEN)
		usleep(USLEEP_INTERVAL);
```

?

`thread_sleep` restarts `nanosleep` if it receives a signal so the loop wouldn't be necessary.

Reply:
> thread_sleep restarts nanosleep
I didn't know about it. Changed to `thread_sleep`.